### PR TITLE
Add examples for documentation

### PR DESCRIPTION
**Pull request by keyneston**
These are mostly examples I wrote for the website proper, but I decided they would fit well in here.

There are a few small changes that fall out side of that.

1. I added a small delay in one of the tests before test cleanup. This was due to it intermittently failing causing the tests to fail.
2. I added a `(*Client).Channel` function which makes it easier to get a channel with known values. It doesn't make an API call in and of itself, just gives you an object that can be used for API calls.

### CODE REVIEW

**Comment by keyneston** [comment]
The test were failing a good portion of the time when it failed to cleanup a channel or channel type during tests. These errors just caused us to run the tests more which then caused more failed cleanup. Either way we end up with junk floating around in the test account. So I changed the cleanup portions of the test to not fail if an error was encountered.

**Comment by tbarbugli** [comment]
> The test were failing a good portion of the time when it failed to cleanup a channel or channel type during tests. These errors just caused us to run the tests more which then caused more failed cleanup. Either way we end up with junk floating around in the test account. So I changed the cleanup portions of the test to not fail if an error was encountered.

IMO the best solution to this problem is to *not* cleanup at all; have tests create data with random IDs and use it only for the current test. Adding a random prefix/suffix to names and IDs is usually the simplest way to keep stuff readable and unique 👉no cleanup.

**Comment by keyneston** [comment]
@tbarbugli we already are doing the random prefix for the names and IDs. We can speed it up by not doing any deletes at all I guess.

File: channel_test.go [hunk 1]
```diff
@@ -411,3 +422,29 @@ func TestChannel_RejectInvite(t *testing.T) {
...
 }
+
+func ExampleChannel_Update() {
+	// https://getstream.io/chat/docs/channel_permissions/?language=python
+	client := &Client{}
+
+	data := map[string]interface{}{
+		"image":      "https://path/to/image",
+		"created_by": "elon",
+		"roles":      map[string]string{"elon": "admin", "gwynne": "moderator"},
```
Comment (thesyncim) [inline channel_test.go]: roles can be a bit confusing (since you are adding a channel and role is a user property )

Comment (keyneston) [reply channel_test.go]: I'm not sure I understand what you are saying. The example was transliterated from the python. So this is just what the python one does 😁

**Pull request merged**